packages/web3/src/web3.ts

```diff
@@ -88,6 +88,17 @@ export class Web3 extends Web3Context<EthExecutionAPI> {
 				} else {
 					super(jsonInterface, self.getContextObject());
 				}
+
+				super.suscribeToGlobalEvents(self);
+			}
+
+			public static set sync_with_globals(value: boolean) {
+				Contract.sync_with_globals = value;
+			}
+
+			public static get sync_with_globals() {
```

Comment:
These shouldn't be snake cased either and I don't think they're needed if we make `syncWithContext` (or `SyncWithGlobals`) a public property of a `Contract` instance configurable via constructor arguments